README.md

```diff
@@ -6,3 +6,16 @@ google-chart
 See https://elements.polymer-project.org/elements/google-chart
 
 [![Build Status](https://travis-ci.org/GoogleWebComponents/google-chart.svg?branch=master)](https://travis-ci.org/GoogleWebComponents/google-chart)
+
+Usage
+=====
+
+The component has to be initialized before using by calling `initializeOnce`
```

Comment:
this seems like a pretty big change? Why does it need to be initialized?

Reply:
Previously, the Google Charts loader script was added in an HTML import and it was a blocking include, i.e. a script tag without async or defer. That guaranteed that `google.charts` namespace is added to the document before the JS code runs.

Polymer 3 uses ES modules and there is no equivalent to the synchronous loading of an external script (unless it is a module as well). Luckily, `<google-chart>` already expects chart constructors to be loaded asynchronously what makes the change very simple.

Introducing such initialization function will also allow additional customizations:

- loading a selected version of Google Charts: e.g. `45.2` or `upcoming` instead of just `current`
- passing API key for country name resolution in GeoChart

Reply:
I support version selection and locale without a function in my 2.1 update:
https://github.com/GoogleWebComponents/google-chart/pull/204/files#diff-d0b4f475ae2630d8efc01169d1a41af6R15

Even without the blocking include, packages were loaded with a promise such that you could make whichever calls you like and they would wait on their package to load before running. I think adding a function to do the initialization is rather cumbersome and kind of an anti-pattern when it comes to Polymer.

Reply:
Yes the packages are already loaded asynchronously. But the loader script is not and if the page is using Google Charts in any other way outside of the `<google-chart>`, you cannot add `<google-chart>` to it (because the loader script can be added only once). Explicit initialization would allow this by doing something like `initializeOnce({googleChartsAlreadyLoader: true});`.

Doing it by a global property also have downsides - see the other comment.

I don't know of any good way to pass such one-time configuration to a web component. The approach with the initialization function was recommended by @rictic.
Is there any?

In some sense it is similar to `<iron-icon>` and `<iron-icons>`. You can do `<iron-icon icon="my-icons:cool-icon">` but it shows nothing until the appropriate iconset is loaded.

Reply:
The best options I know of for doing global configuration are: 
  - set globals before the first import, which can be a tricky constraint to maintain in your build system
  -  add an initialization function that must be called exactly once

The former is nice because you can have defaults, so that it's easier to use out of the box, but the downside is that you have to be super careful everywhere that you use the element that you first set the globals (and that your build and test tooling maintains the order of your code, which some do not by default).

Reply:
@wesalvaro: what do you think?

Reply:
I'm just not seeing any benefit from the `initializeOnce()` approach.

- Versions other than `current` shouldn't need to be set.
   If they do, it should be _global_ and setting it in the document's head seems fine.
   There isn't much build system input to it.
- Allowing a function call to set versions other than `current` seems like it invites issues.
   e.g. Calling more than once to make sure it's called throws.
- The loader can always check if the Charts library is already loaded before loading.
   This way, the user can load the chart loader however they want.

Reply:
We can avoid the `initializeOnce` function and simplify the default case if we change the component to load Google Chart only if it has not been loaded already: see #225.
I think we can merge that PR to the current master. Then converting the component to Polymer 3 should be as simple as running the modulizer.